### DISCUSSION TITLE
Tailwind's classname overriding depends on the sequence on config file

### DISCUSSION THREAD

**Discussion by jasbindar-singh** [tailwindlabs/tailwindcss#4861] [Help]
So I created a Button component where I wanted the user to add class names on top of existing class names, here's my Button component,
```
const Button = ({ children, className = '' }) => {
  const defaultClassNames = "text-color-primary p-16"
  const classNames = `${defaultClassNames} ${className}`
  return <button className={classNames}>{children}</button>
}
```
and this was my config
```
...
extends: {
  color: {
    white: '#fff',
    primary: '#666',
    black: '#000'
  },
  spacing: {
    12: '12px',
    16: '16px',
    21: '21px'
  }
}
```
While using my Button component and overriding class names I noticed something weird, when I used it,
```
<Button className="p-21 text-color-black">Login</Button>
```
This used to work, the class names I passed took precedence as expected, but when I tried this,
```
<Button className="p-12 text-color-white">Login</Button>
```
The class names that I passed did not override the default class names.

Is this the expected behavior, or am I missing something? How do I override the class names that are before the used values in the config?

**Comment by dance2die** [comment]
Looks like the configuration is malformed.

1. Change `extends` => `extend`
2. Change `color` => `colors`

Configuration Doc: https://tailwindcss.com/docs/customizing-colors#extending-the-defaults

```diff
-extends: {
+extend: {
-  color: {
+ colors: {
    white: '#fff',
    primary: '#666',
    black: '#000'
  }
}
```

You can see the working demo here: https://stackblitz.com/edit/button-issue?file=pages%2Findex.js
